test/app-luatest/gh_8613_new_cli_behaviour_test.lua

```diff
@@ -0,0 +1,106 @@
+local t = require('luatest')
+local treegen = require('test.treegen')
+local justrun = require('test.justrun').tarantool
+
+local g = t.group()
+
+local misuse = {
+    output = "Invalid usage:"
+          .. " please either provide a Lua script name"
+          .. " or specify an instance name to be started"
+          .. " or set -i CLI flag to spawn Lua REPL.",
+    pattern = false,
+}
+local greeting = {
+    output = "Tarantool [^\n]*\ntype 'help' for interactive help",
+    pattern = true,
+}
+local prompt = {
+    output = 'tarantool>',
+    pattern = false,
+}
+
+local script = string.dump(function() print(_VERSION) end)
+
+local rundir
+g.before_each(function()
+    treegen.init(g)
```

Comment:
There should be a `treegen.cleanup()` call somewhere (`g.after_all()` or `g.after_each()`). Otherwise the temporary directories will stay there.